protocol/nats_jetstream/v2/message.go

```diff
@@ -8,13 +8,26 @@ package nats_jetstream
 import (
 	"bytes"
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/nats-io/nats.go"
 
 	"github.com/cloudevents/sdk-go/v2/binding"
 	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
 )
 
+const (
+	_EMPTY_           = ""
+	prefix            = "ce-"
+	contentTypeHeader = "content-type"
+	// see https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/nats-protocol-binding.md
+	applicationCloudEvent = "application/cloudevents"
```

Comment:
I wonder why `application/cloudevents+json` is not used for structured mode in the binding proposal?

Reply:
Not sure.  I was following the proposal document.  If you like, I can change this value and the proposal document.

Reply:
Re-posting the question, in case it was missed:
Would you like me to change this to `application/cloudevents+json`?  Would you also like me to change the proposal to match?

Reply:
IMHO that would make sense to clearly distinguish between the two transport modes. It is also used in the Kafka protocol [implementation](https://github.com/cloudevents/sdk-go/blob/7f5ef3992769f96b40a54c4d59291be62acd36da/protocol/kafka_sarama/v2/message_test.go#L30)

Reply:
After thinking about this, those who might be using structured mode might not be setting NATS headers at all, including the content-type.  I figure those using binary mode do in fact have to set NATS headers, including the required cloudEvent headers.